From fc003fe211ae03e911c0a4d1f823788bbbce1900 Mon Sep 17 00:00:00 2001
From: Nick Kossifidis <mickflemm@gmail.com>
Date: Mon, 29 Oct 2018 12:36:18 +0200
Subject: [PATCH 4/5] Use the generic fcntl.h header Currently the fcntl.h

header uses different macros from those on upstream Linux. On upstream Linux there is no arch-specific fcntl.h, the generic one is used. As a result various syscalls using these macros fail, an example being BusyBox where any call to opendir() will fail because O_DIRECTORY id is different between the headers used here (probably coming from arm64) and the generic headers used by the kernel. Simple delete the arch-specific fcntl.h and let musl use the generic header as well.
---
 arch/riscv32/bits/fcntl.h | 38 --------------------------------------
 arch/riscv64/bits/fcntl.h | 38 --------------------------------------
 2 files changed, 76 deletions(-)
 delete mode 100644 arch/riscv32/bits/fcntl.h
 delete mode 100644 arch/riscv64/bits/fcntl.h

diff --git a/arch/riscv32/bits/fcntl.h b/arch/riscv32/bits/fcntl.h
deleted file mode 100644
index 92787976..00000000
--- a/arch/riscv32/bits/fcntl.h
+++ /dev/null
@@ -1,38 +0,0 @@
-#define O_CREAT        0100
-#define O_EXCL         0200
-#define O_NOCTTY       0400
-#define O_TRUNC       01000
-#define O_APPEND      02000
-#define O_NONBLOCK    04000
-#define O_DSYNC      010000
-#define O_SYNC     04010000
-#define O_RSYNC    04010000
-#define O_DIRECTORY  040000
-#define O_NOFOLLOW  0100000
-#define O_CLOEXEC  02000000
-
-#define O_ASYNC      020000
-#define O_DIRECT    0200000
-#define O_LARGEFILE 0400000
-#define O_NOATIME  01000000
-#define O_PATH    010000000
-#define O_TMPFILE 020040000
-#define O_NDELAY O_NONBLOCK
-
-#define F_DUPFD  0
-#define F_GETFD  1
-#define F_SETFD  2
-#define F_GETFL  3
-#define F_SETFL  4
-#define F_GETLK  5
-#define F_SETLK  6
-#define F_SETLKW 7
-#define F_SETOWN 8
-#define F_GETOWN 9
-#define F_SETSIG 10
-#define F_GETSIG 11
-
-#define F_SETOWN_EX 15
-#define F_GETOWN_EX 16
-
-#define F_GETOWNER_UIDS 17
diff --git a/arch/riscv64/bits/fcntl.h b/arch/riscv64/bits/fcntl.h
deleted file mode 100644
index 92787976..00000000
--- a/arch/riscv64/bits/fcntl.h
+++ /dev/null
@@ -1,38 +0,0 @@
-#define O_CREAT        0100
-#define O_EXCL         0200
-#define O_NOCTTY       0400
-#define O_TRUNC       01000
-#define O_APPEND      02000
-#define O_NONBLOCK    04000
-#define O_DSYNC      010000
-#define O_SYNC     04010000
-#define O_RSYNC    04010000
-#define O_DIRECTORY  040000
-#define O_NOFOLLOW  0100000
-#define O_CLOEXEC  02000000
-
-#define O_ASYNC      020000
-#define O_DIRECT    0200000
-#define O_LARGEFILE 0400000
-#define O_NOATIME  01000000
-#define O_PATH    010000000
-#define O_TMPFILE 020040000
-#define O_NDELAY O_NONBLOCK
-
-#define F_DUPFD  0
-#define F_GETFD  1
-#define F_SETFD  2
-#define F_GETFL  3
-#define F_SETFL  4
-#define F_GETLK  5
-#define F_SETLK  6
-#define F_SETLKW 7
-#define F_SETOWN 8
-#define F_GETOWN 9
-#define F_SETSIG 10
-#define F_GETSIG 11
-
-#define F_SETOWN_EX 15
-#define F_GETOWN_EX 16
-
-#define F_GETOWNER_UIDS 17
-- 
2.20.1

